### DISCUSSION TITLE
Negative values in energy dashboard

### DISCUSSION THREAD

**Discussion by bb12489** [BJReplay/ha-solcast-solar#75] [General]
Not sure why I'm seeing this all of a sudden, but I'm getting a negative value for the forecast on my energy dashboard. I've removed the integration completely and reinstalled without any success in fixing this. I'm on 4.0.36
![image](https://github.com/BJReplay/ha-solcast-solar/assets/13389071/997591e7-beb0-4453-a4d3-0bb65c5740a8)

**Comment by autoSteve** [comment]
What's the config of the energy dashboard for solar production?

You should have a single solar export sensor (assuming you get production value from a single source), with a 'forecast production' selected and Solcast checked.

The only other time I've seen this kind of weirdness is where there is a separate production sensor configured referencing something forecast related as the solar production entity, which makes no sense. This seems likely in your case because the bars are of differing brown colours, indicating two production entities... This will result in garbage in, garbage out if one of them is a forecast of actual production.

<img width="443" alt="image" src="https://github.com/BJReplay/ha-solcast-solar/assets/37229860/d0166b83-b322-43ec-8f1a-a28670950197">

**Comment by gcoan** [comment]
> Not sure why I'm seeing this all of a sudden, but I'm getting a negative value for the forecast on my energy dashboard. I've removed the integration completely and reinstalled without any success in fixing this. I'm on 4.0.36
> 
> ![image](https://github.com/BJReplay/ha-solcast-solar/assets/13389071/997591e7-beb0-4453-a4d3-0bb65c5740a8)
> 
> 

It appears that your generation is showing negative values, those are the solid bars.  The forecast (from Solcast) is the dashed line (that I can't really see)

**Reply by autoSteve** [reply]
Plus the different brown colour for clearly two generation values. This is a messed up configuration. I want to know which entities are configured for "Solar production energy". There are definitely two in this case...

A cry of, "All of a sudden" I have heard before because transitioning from Oziee to BJReplay, and that turned out to be a whacky config used for Oziee. One could posture that Oziee's version was "relaxed" in regard to what all is happening here, and that we are now seemingly not, but not much has changed with regards to sensor values... They are what they are, and they are pretty much the same... We have embraced/extended, and have gone nowhere near a re-write.

**Comment by autoSteve** [comment]
@bb12489, answers to the following questions are vital if you want any further reponse.

* What version were you running prior, after which this 'suddenly' appeared? (We know you're on the latest 4.0.36 beta, and good on you. It's great code. Fourty-eight souls so far, excluding yourself have not seen anything like this. And eighty-four for the prior 4.0.35, with not a whisper. So I think we can rule out version.)
* What is the configuration of your Energy Dashboard solar sensors?

Without that detail anyone can only guess.

Remove and re-install _will_ give the same answer. Don't bother with that, as it's just wasting your time. Code is code. It does what it does. It will give the same answer in this case at the same point in time as it did a moment earlier.

I'm far more interested in what your Energy dashboard config is. And I'm also far more interested if you revert to the prior version that you were running does the issue disappear? (Which would be fascinating...)

**Comment by autoSteve** [comment]
This discussion and possible help will be shut down and closed in precisely 12 hours if no useful information is received.

There are two production sensors at play here, as shown in the chart. These are not relevant to a Solcast forecast code issue, and likely a configuration issue.

**Comment by bb12489** [comment]
Apologies for the late reply here. I can assure you that I only have one forecast sensor in my config (shown below). I initially setup the Solcast integration before it changed hands. All I've done since then is switch to this new repo for future updates, and just recently updated to 4.0.36 to see if it would fix this negative forecast value issue. I've also tried downgrading to previous versions without any luck. So yes...this issue "all of sudden" happened without any configuration change to my sensors. 

![image](https://github.com/BJReplay/ha-solcast-solar/assets/13389071/30f3b3e2-fe4d-465d-bdec-761338547e1b)

![image](https://github.com/BJReplay/ha-solcast-solar/assets/13389071/21e51f83-d0fd-4cd8-8341-67d7207b51e1)

![image](https://github.com/BJReplay/ha-solcast-solar/assets/13389071/47b0d61d-a015-4265-b0e1-217a523d8763)

![image](https://github.com/BJReplay/ha-solcast-solar/assets/13389071/d832c3bc-e504-40df-b47d-8d207daa0357)

**Reply by autoSteve** [reply]
Great info.

You have two production sensors configured, and this is the cause. How this was working earlier is anyone's guess, but I have seen an incorrect set up like this before.

Get rid of the `Solar PV Forecast Forecast Today` one, as that makes no sense as a production sensor entity, and Select "Forecast production" as Solcast for `sensor.net_solar_production integral`. This will fix the production chart.

Cheers.

**Comment by bb12489** [comment]
That seems to have done it. Thanks for the help.

**Reply by autoSteve** [reply]
Apologies for an earlier comment. I meant to say "Two **production** sensors configured", and not forecast sensors. That was misleading, so I corrected that.
